lib/src/client/transport/xhr_transport.dart

```diff
@@ -162,10 +177,26 @@ class XhrClientConnection extends ClientConnection {
 
   @override
   GrpcTransportStream makeRequest(String path, Duration timeout,
-      Map<String, String> metadata, ErrorHandler onError) {
+      Map<String, String> metadata, ErrorHandler onError,
+      {CallOptions callOptions}) {
+    // gRPC-web headers.
+    if (_getContentTypeHeader(metadata) == null) {
+      metadata['Content-Type'] = 'application/grpc-web+proto';
+      metadata['X-User-Agent'] = 'grpc-web-dart/0.1';
+      metadata['X-Grpc-Web'] = '1';
+    }
+
```

Comment:
I suggest doing:

```dart
Uri requestUri = uri.resolve(path);
```

already here. Then below you can do 

```dart
requestUri = cors.bypassCorsPreflight(metadata, requestUri)
```

which would simplify parameter manipulation code considerably.

Reply:
Done.